index.d.ts

```diff
@@ -126,6 +141,7 @@ export function parse(query: string, options?: ParseOptions): ParsedQuery;
 export interface ParsedUrl {
 	readonly url: string;
 	readonly query: ParsedQuery;
+	readonly fragmentIdentifier?: string;
```

Comment:
You could make `ParsedUrl` generic so `fragmentIdentifier` only exist when the `parseFragmentIdentifier` option is `true`. That would be nicer for user as it wouldn't be an optional.

Reply:
@sindresorhus I didn't get get what you meant. Can you give an example

Reply:
I was thinking something similar with https://github.com/sindresorhus/query-string/pull/222/files#diff-b52768974e6bc0faccb7d4b75b162c99R152, where the option value controls what's returned, but after some more thinking, I don't think it's worth complicating it.